pkg/gofer/origins/origin.go

```diff
@@ -96,41 +161,51 @@ func NewSet(list map[string]Handler) *Set {
 	return &Set{list: list}
 }
 
-func DefaultSet() *Set {
-	const defaultWorkerCount = 5
-	httpWorkerPool := query.NewHTTPWorkerPool(defaultWorkerCount)
-
+func DefaultSet(httpWorkerPool query.WorkerPool) *Set {
 	return NewSet(map[string]Handler{
-		"balancer":      &Balancer{Pool: httpWorkerPool},
-		"binance":       &Binance{Pool: httpWorkerPool},
-		"bitfinex":      &Bitfinex{Pool: httpWorkerPool},
-		"bitstamp":      &Bitstamp{Pool: httpWorkerPool},
-		"bitthumb":      &BitThump{Pool: httpWorkerPool},
-		"bithumb":       &BitThump{Pool: httpWorkerPool},
-		"bittrex":       &Bittrex{Pool: httpWorkerPool},
-		"coinbasepro":   &CoinbasePro{Pool: httpWorkerPool},
-		"cryptocompare": &CryptoCompare{Pool: httpWorkerPool},
-		"coinmarketcap": &CoinMarketCap{Pool: httpWorkerPool, APIKey: "KEY_HERE"},
-		"ddex":          &Ddex{Pool: httpWorkerPool},
-		"folgory":       &Folgory{Pool: httpWorkerPool},
-		"ftx":           &Ftx{Pool: httpWorkerPool},
-		"fx":            &Fx{Pool: httpWorkerPool},
-		"gateio":        &Gateio{Pool: httpWorkerPool},
-		"gemini":        &Gemini{Pool: httpWorkerPool},
-		"hitbtc":        &Hitbtc{Pool: httpWorkerPool},
-		"huobi":         &Huobi{Pool: httpWorkerPool},
-		"kraken":        &Kraken{Pool: httpWorkerPool},
-		"kucoin":        &Kucoin{Pool: httpWorkerPool},
-		"kyber":         &Kyber{Pool: httpWorkerPool},
-		"loopring":      &Loopring{Pool: httpWorkerPool},
-		"okex":          &Okex{Pool: httpWorkerPool},
-		"poloniex":      &Poloniex{Pool: httpWorkerPool},
-		"sushiswap":     &Sushiswap{Pool: httpWorkerPool},
-		"uniswap":       &Uniswap{Pool: httpWorkerPool},
-		"upbit":         &Upbit{Pool: httpWorkerPool},
+		"balancer":          NewHandler("balancer", httpWorkerPool, nil),
```

Comment:
How about keeping the `DefaultSet` as it was previously - without any exchanges that require an api key - and then creating a new one that has the defaults as the base with additional exchanges that do require more params.